docs/compiling.rst

```diff
@@ -13,6 +15,129 @@ the [python_example]_ repository.
 
 .. [python_example] https://github.com/pybind/python_example
 
+A helper file is provided with pybind11 that can simplify usage with setuptools.
+
+To use pybind11 inside your ``setup.py``, you have to have some system to
+ensure that ``pybind11`` is installed when you build your package. There are
+four possible ways to do this, and pybind11 supports all four: You can ask all
+users to install pybind11 beforehand (bad), you can use
```

Comment:
The intention was to highlight that doing that is bad, as it might be the first thought for someone just getting started.  I've seen quite a bit of this in Physics, at least, where packages are literally impossible to put into a Pipfile since they are not installed sequentially.

Reply:
You've seen Math/Physics people use Python packaging? Remarkable... 😏.

Most of these comments are merely suggestions, while there might be questions about the approach, they won't be able to say this PR isn't well polished.